### PR TITLE
(consul & google) enable/disable instance

### PR DESCRIPTION
**Pull request by lwander**
@duftler

### CODE REVIEW

**Comment by lwander** [comment]
ping @duftler

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/converters/discovery/EnableInstancesInDiscoveryConverter.groovy [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016 Netflix, Inc.
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/converters/discovery/EnableInstancesInDiscoveryConverter.groovy]: s/Netflix/Google

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/description/GoogleInstanceListDescription.groovy [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.google.deploy.description
+
+class GoogleInstanceListDescription extends AbstractGoogleCredentialsDescription {
+  String asgName
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/description/GoogleInstanceListDescription.groovy]: s/asgName/serverGroupName

Will need to ensure this attribute is being passed from orca.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/discovery/AbstractEnableDisableInstancesInDiscoveryOperation.groovy [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    instances.each { String instance ->
+      task.updateStatus phaseName, "$presentParticipling instance $instance..."
+      EnableDisableConsulInstance.operate(credentials.consulConfig,
+        instance,
+        disable
+          ? EnableDisableConsulInstance.State.disable
+          : EnableDisableConsulInstance.State.enable)
+    }
+
+    return
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/discovery/AbstractEnableDisableInstancesInDiscoveryOperation.groovy]: If we're going to have the `return`, might as well have it return `null`.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/discovery/DisableInstancesInDiscoveryOperation.groovy [hunk 4]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2016 Netflix, Inc.
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/discovery/DisableInstancesInDiscoveryOperation.groovy]: s/Netflix/Google

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/discovery/DisableInstancesInDiscoveryOperation.groovy [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  @Override
+  boolean isDisable() {
+    return false
+  }
+
+  @Override
+  String getPhaseName() {
+    return "ENABLE_INSTANCE_DISCOVERY"
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/discovery/DisableInstancesInDiscoveryOperation.groovy]: s/ENABLE_INSTANCE/DISABLE_INSTANCES

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/discovery/EnableInstancesInDiscoveryOperation.groovy [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  @Override
+  boolean isDisable() {
+    return false
+  }
+
+  @Override
+  String getPhaseName() {
+    return "ENABLE_INSTANCE_DISCOVERY"
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/discovery/EnableInstancesInDiscoveryOperation.groovy]: s/INSTANCE/INSTANCES

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/validators/AbstractEnableDisableInstancesInDiscoveryDescriptionValidator.groovy [hunk 7]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016 Netflix, Inc.
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/validators/AbstractEnableDisableInstancesInDiscoveryDescriptionValidator.groovy]: s/Netflix/Google

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/validators/AbstractEnableDisableInstancesInDiscoveryDescriptionValidator.groovy [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+abstract class AbstractEnableDisableInstancesInDiscoveryDescriptionValidator extends DescriptionValidator<GoogleInstanceListDescription> {
+  @Autowired
+  AccountCredentialsProvider accountCredentialsProvider
+
+  @Override
+  void validate(List priorDescriptions, GoogleInstanceListDescription description, Errors errors) {
+    def helper = new StandardGceAttributeValidator("googleInstanceListDescription", errors)
+
+    helper.validateNotEmpty(description.instanceIds, "instanceIds")
+    helper.validateRegion(description.region)
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/validators/AbstractEnableDisableInstancesInDiscoveryDescriptionValidator.groovy]: You'll need to add the `description.credentials` argument due to a PR that crossed in the mail.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/validators/DisableInstancesInDiscoveryDescriptionValidator.groovy [hunk 9]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2016 Netflix, Inc.
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/validators/DisableInstancesInDiscoveryDescriptionValidator.groovy]: s/Netflix/Google

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/validators/EnableInstancesInDiscoveryDescriptionValidator.groovy [hunk 10]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2016 Netflix, Inc.
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/validators/EnableInstancesInDiscoveryDescriptionValidator.groovy]: s/Netflix/Google

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/description/GoogleInstanceListDescription.groovy (hunk 2 above)
Comment (lwander) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/description/GoogleInstanceListDescription.groovy]: It's initially written by Deck, I'll add it there.

**Pull request merged**